### ISSUE TITLE
Add Parser 3.0.0.0 support

### ISSUE THREAD

**Issue by bkuhlmann** [troessner/reek#1576]
## Overview

You might already be aware of this but with the release of Rubocop 1.8.0, the Rubocop team has required the use of Parser 3.0.0.0. Would it be possible to upgrade Reek to support Parser 3.0.0.0 so one can use both Rubocop and Reek in the same project?

## Screenshots/Screencasts

![iTerm2-he-deployer-VioeQq6x](https://user-images.githubusercontent.com/50245/104046863-f9f17b80-519d-11eb-8dad-10f2af60b88f.png)

## Steps to Recreate

Save the following as `snippet.rb` and run as `ruby snippet.rb`:

``` ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile true do
  source "https://rubygems.org"
  gem "reek", "~> 6.0"
  gem "rubocop", "~> 1.8"
end

puts "This script will error."
```

## Desired Behavior

Would be awesome to upgrade to Rubocop 1.8.0 while still using Reek.

## Environment

``` 
rubocop -V

1.7.0 (using Parser 2.7.2.0, rubocop-ast 1.4.0, running on ruby 3.0.0 x86_64-darwin20)
  - rubocop-performance 1.9.2
  - rubocop-rake 0.5.1
  - rubocop-rspec 2.1.0
```

**Comment by mvz** [comment]
Normally, Dependabot would have taken care of this already, but it seems it messed up: https://github.com/troessner/reek/pull/1572/files#diff-c08372593ec972b81a370fcef9ff665a5ad090dc230e814bfabb32007b617227R31

**Comment by bkuhlmann** [comment]
@mvz Ha, yeah...I suppose those requirements are a little hard to decipher for Dependabot too. 😅

**Comment by troessner** [comment]
Just released Reek 6.0.3 with the request parser dependency upgrade @bkuhlmann  ;)

**Comment by bkuhlmann** [comment]
:bow: @troessner, this is fantastic! I was working on a feature branch over the weekend where I was doing this:

``` 
gem "reek", github: "troessner/reek", ref: "cac6f0193b95fca97e1f2f155ac679ce2b765ab5"
```

With a Rebase fixup, I'll make that look like I was using `6.0.3` all along. Ha! :wink:
